client/state/themes/selectors/is-theme-woo-commerce.ts

```diff
@@ -10,5 +10,8 @@ export function isThemeWooCommerce( state: IAppState, themeId: string ) {
 	const theme = getTheme( state, 'wpcom', themeId );
 	const themeSoftwareSetTaxonomy: ThemeSoftwareSet[] | undefined =
 		theme?.taxonomies?.theme_software_set;
-	return themeSoftwareSetTaxonomy?.some( ( { slug } ) => slug === 'woo-on-plan' );
+	return (
```

Comment:
Fixes: 

1. The actual slug used is `woo-on-plans` for WooCommerce themes
2. Replace deconstructor with optional chaining `slug` to be more defensive
3. Guarantee a boolean return by adding `|| false`

cc @okmttdhr in case you have concerns